source/server/listener_manager_impl.cc

```diff
@@ -379,11 +374,33 @@ std::vector<std::reference_wrapper<Listener>> ListenerManagerImpl::listeners() {
   return ret;
 }
 
+void ListenerManagerImpl::addListenerToWorker(Worker& worker, ListenerImpl& listener) {
+  worker.addListener(listener, [this, &listener](bool success) -> void {
+    // The add listener completion runs on the worker thread. Post back to the main thread to
+    // avoid locking.
+    server_.dispatcher().post([this, success, &listener]() -> void {
+      // It is theoretically possible for a listener to get added on 1 worker but not the others.
+      // The following code will yield 1 critical log and 1 stat. It will also cause listener
```

Comment:
I would rephrase this as it kind of confused me on the first pass, something to the effect of "The below check with onListenerCreateFailure() is there to ensure we execute the removal/logging/stats at most once on failure."